JSONModel/JSONModel/JSONModel.m

```diff
@@ -841,22 +861,30 @@ -(id)__reverseTransform:(id)value forProperty:(JSONModelClassProperty*)property
 }
 
 #pragma mark - custom transformations
-- (BOOL)__customSetValue:(id <NSObject>)value forProperty:(JSONModelClassProperty *)property
+- (BOOL)__customSetValue:(id <NSObject>)value forProperty:(JSONModelClassProperty *)property error:(NSError**) error
 {
     NSString *class = NSStringFromClass([JSONValueTransformer classByResolvingClusterClasses:[value class]]);
 
-    SEL setter = nil;
-    [property.customSetters[class] getValue:&setter];
+    JSONModelCustomSetter* customSetter = property.customSetters[class];
 
-    if (!setter)
-        [property.customSetters[@"generic"] getValue:&setter];
+    if (!customSetter)
+        customSetter = property.customSetters[@"generic"];
 
-    if (!setter)
+    if (!customSetter)
         return NO;
 
+    SEL setter = nil;
+    [customSetter.value getValue:&setter];
+
     IMP imp = [self methodForSelector:setter];
-    void (*func)(id, SEL, id <NSObject>) = (void *)imp;
-    func(self, setter, value);
+    
+    if(customSetter.withErrorOutParam){
+        void (*func)(id, SEL, id <NSObject>, NSError**) = (void *)imp;
```

Comment:
Should have a BOOL response, e.g.:

```objc
- (BOOL)setFooWithNSString:(NSString *)str error:(NSError **)error
{
  *error = [NSError blah];
  return NO;
}
```

Should throw an exception if implementation looks like this:

```objc
- (void)setFooWithNSString:(NSString *)str error:(NSError **)error;
```

Not sure if this can be checked dynamically?